torch/_refs/nn/functional/__init__.py

```diff
@@ -262,6 +263,37 @@ def relu(a: TensorLikeType, inplace: bool = False) -> TensorLikeType:
     return torch.where(torch.le(a, 0), 0, a)
 
 
+@register_decomposition(aten.channel_shuffle)
+@aten.channel_shuffle.default.py_impl(DispatchKey.CompositeImplicitAutograd)
+@aten.channel_shuffle.default.py_impl(DispatchKey.Autograd)
+@out_wrapper()
+def channel_shuffle(input: TensorLikeType, groups: int) -> TensorLikeType:
+    """
+    Reference implementation of :func:`torch.nn.functional.channel_shuffle`.
+    """
+    from torch._meta_registrations import device_hint
+
+    torch._check(
+        groups > 0,
+        lambda: f"Number of groups to divide channels in must be positive. Value of groups:{groups}",
+    )
+    n = input.shape[0]
+    c = input.shape[1]
+    cg = c // groups
+    dhw = input.shape[2:]
+
+    if input.numel() == 0 or (
+        device_hint(input) == "cuda" and (groups == 1 or groups == c)
+    ):
+        return input.view(input.shape)
```

Comment:
Views should be represented in the meta function, so it would be simpler to just delete the meta registration and let the decomposition act as the meta function.